libs/application/templates/public-debt-payment-plan/src/lib/paymentPlanUtils.ts

```diff
@@ -23,3 +29,68 @@ export const prerequisitesFailed = (data: ExternalData) => {
 
 export const formatIsk = (value: number): string =>
   value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, '.') + ' kr.'
+
+export const useDistributionTable = ({
+  monthAmount,
+  monthCount,
+  totalAmount,
+  scheduleType,
+}: PaymentDistribution) => {
+  const getDistribution = useLazyDistribution()
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [
+    distributionData,
+    setDistributionData,
+  ] = useState<PaymentScheduleDistribution | null>(null)
+
+  const getDistributionCallback = useCallback(
+    async ({
+      monthAmount = null,
+      monthCount = null,
+      totalAmount,
+      scheduleType,
+    }: GetScheduleDistributionInput) => {
+      const { data } = await getDistribution({
+        input: {
+          monthAmount,
+          monthCount,
+          totalAmount,
+          scheduleType,
+        },
+      })
+
+      return data
+    },
+    [getDistribution],
+  )
+
+  useEffect(() => {
+    setIsLoading(true)
+    getDistributionCallback({
+      monthAmount,
+      monthCount,
+      totalAmount,
+      scheduleType,
+    })
+      .then((response) => {
+        if (!response) return
```

Comment:
Should this be handled somehow to show the user that no response was received?

Reply:
I'm going to remove this. I don't think this should ever fail. But there is a lot of rethinking that needs to take place when people come from summer break when it comes to the payment plan.